### ISSUE TITLE
Simple job hangs

### ISSUE THREAD

**Issue by gobbedy** [pgiri/dispy#134]
Hello,

This is my first time using dispy and it hangs with this simple example. I'm wondering if you can point me in the right direction to fix this.

```python

import dispy
def compute():
    return 0

cluster = dispy.JobCluster(compute)
job = cluster.submit(1)
print("NOTHING PRINTED AFTER THIS")
host, n = job() # waits for job to finish and returns results
print("NOT PRINTED")
cluster.print_status()
```

This is what the above testcase prints:

> 2018-07-17 19:13:34 pycos - version 4.7.5 with epoll I/O notifier
> 2018-07-17 19:13:34 dispy - dispy client version: 4.9.0
> 2018-07-17 19:13:34 dispy - Storing fault recovery information in "_dispy_20180717191334"
> NOTHING PRINTED AFTER THIS

When I do a keyboard interrupt, I see that it's just stuck waiting in the threading module:

>   File "/path/to/install/lib/python3.6/site-packages/dispy/__init__.py", line 124, in __call__
>     self.finish.wait()
>   File "/path/to/install/python/3.6.3/lib/python3.6/threading.py", line 551, in wait
>     signaled = self._cond.wait(timeout)
>   File "/path/to/install/python/3.6.3/lib/python3.6/threading.py", line 295, in wait
>     waiter.acquire()

If it helps, I'm on a Cluster using CentOS 7, and using python 3.6.3

**Comment by pgiri** [comment]
It seems scheduler hasn't found any server nodes. Make sure `dispynode.py` is running (e.g., you can start one on the same computer running client in a different terminal).

**Comment by gobbedy** [comment]
@pgiri  Thanks!  That was it, just needed to launch dispynode.py. I launched it locally as you suggested and it immediately worked. I also tested on the remote server adding the "nodes" argument and it also worked.

Thanks for catching the typos too, I was too quick in simplifying the official example.

Regarding the hanging, should it spit out an error instead of hanging when it has no servers to connect to? This might help debug the issue.

Also, for future total noobs like me, would it be a good idea to publish a basic how-to? eg 
1. launch `dispynode.py` on server
2. replace nodename1 in `example.py` with hostname of server in step 1
3. launch `example.py` on client

where example.py might be [this excellent example](http://dispy.sourceforge.net/examples.html) with this line  
`cluster = dispy.JobCluster(compute, depends=[C])`
replaced with this  
`cluster = dispy.JobCluster(compute, depends=[C], nodes=[nodename1,])`
